plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/model/ControlFlowEdge.java

```diff
@@ -36,7 +36,7 @@ public ControlFlowEdge(Node start, Node end) {
 	 */
 	public ControlFlowEdge(Node start, Node end, ControlFlowType cfType) {
 		super(start, end);
-		assert cfType.isBackwards() || start.astPosition < end.astPosition : "Edge has wrong direction";
+		assert cfType.isBackwards() == start.astPosition > end.astPosition : "Edge has wrong direction";
```

Comment:
a bit of nesting here and operator precedence is happening here. Could you add some `(` and `)` ? Like one pair at least?

Reply:
done